src/SerializeAST.cc

```diff
@@ -0,0 +1,766 @@
+
+    return typeSpecificationObject;
+}
+
+sos::Array WrapTypeAttributes(const mson::TypeAttributes& typeAttributes)
+{
+    sos::Array typeAttributesArray;
+
+    if (typeAttributes & mson::RequiredTypeAttribute) {
+        typeAttributesArray.push(sos::String("required"));
+    }
+    else if (typeAttributes & mson::OptionalTypeAttribute) {
+        typeAttributesArray.push(sos::String("optional"));
+    }
+    else if (typeAttributes & mson::DefaultTypeAttribute) {
+        typeAttributesArray.push(sos::String("default"));
+    }
+    else if (typeAttributes & mson::SampleTypeAttribute) {
+        typeAttributesArray.push(sos::String("sample"));
+    }
+    else if (typeAttributes & mson::FixedTypeAttribute) {
+        typeAttributesArray.push(sos::String("fixed"));
+    }
+
+    return typeAttributesArray;
+}
+
+sos::Object WrapTypeDefinition(const mson::TypeDefinition& typeDefinition)
+{
+    sos::Object typeDefinitionObject;
+
+    // Type Specification
+    typeDefinitionObject.set(SerializeKey::TypeSpecification, WrapTypeSpecification(typeDefinition.typeSpecification));
+
+    // Type Attributes
+    typeDefinitionObject.set(SerializeKey::Attributes, WrapTypeAttributes(typeDefinition.attributes));
+
+    return typeDefinitionObject;
+}
+
+sos::Object WrapValueDefinition(const mson::ValueDefinition& valueDefinition)
+{
+    sos::Object valueDefinitionObject;
+
+    // Values
+    valueDefinitionObject.set(SerializeKey::Values,
+                              WrapCollection<mson::Value>()(valueDefinition.values, WrapValue));
+
+    // Type Definition
+    valueDefinitionObject.set(SerializeKey::TypeDefinition, WrapTypeDefinition(valueDefinition.typeDefinition));
+
+    return valueDefinitionObject;
+}
+
+sos::Object WrapPropertyName(const mson::PropertyName& propertyName)
+{
+    sos::Object propertyNameObject;
+
+    if (!propertyName.literal.empty()) {
+        propertyNameObject.set(SerializeKey::Literal, sos::String(propertyName.literal));
+    }
+    else if (!propertyName.variable.empty()) {
+        propertyNameObject.set(SerializeKey::Variable, WrapValueDefinition(propertyName.variable));
+    }
+
+    return propertyNameObject;
+}
+
+// Forward declarations
+sos::Object WrapTypeSection(const mson::TypeSection& typeSection);
+
+
+const sos::String TypeSectionToString(const mson::TypeSection& section)
+{
+    switch (section.klass) {
+        case mson::TypeSection::BlockDescriptionClass:
+            return sos::String("blockDescription");
+
+        case mson::TypeSection::MemberTypeClass:
+            return sos::String("memberType");
+
+        case mson::TypeSection::SampleClass:
+            return sos::String("sample");
+
+        case mson::TypeSection::DefaultClass:
+            return sos::String("default");
+
+        default:
+            return sos::String();
+    }
+    return sos::String();
+}
+
+sos::String AssetRoleToString(const AssetRole& role)
```

Comment:
I change name, due this just only translate type to string. Wrap I keek just functions which handle some kind of object, not for primitives